npm/private/npm_translate_lock.bzl

```diff
@@ -331,6 +335,18 @@ def _gen_npm_imports(lockfile, attr):
             friendly_name not in attr.lifecycle_hooks_exclude
         )
 
+        url = None
+        if tarball:
+            if _is_url(tarball):
+                url = tarball
+            else:
+                # pnpm will omit the registry component from the tarball value when it is configured
+                # via an .npmrc registry setting for the package. We prefix with the npm registry
```

Comment:
Why don't we read .npmrc to get a correct answer? Todo for that ?

Reply:
Then print a warning and suggest upgrade?

Reply:
✅